test_interpreter.py

```diff
@@ -0,0 +1,311 @@
+from __future__ import with_statement
```

Comment:
These tests are looking pretty good! My only comments are:

*  the test bytecode programs are too large. Ideally you want the smallest programs you can build to test individual byte-codes. This will really help you to isolate issues, as when something fails there will only be a handful of byte-codes that could be the issue. In larger programs, the problem will be much harder to find.

* You may find it easier to put all of the support code into a file, and then have each test file subclass it.